httplib.h

```diff
@@ -5596,7 +5596,7 @@ inline bool Server::listen_internal() {
 #if __cplusplus > 201703L
```

Comment:
If your change is applied, the `#if __cplusplus > 201703L` block seems to be no longer needed. Could you make remove the block and check with C++17 compile option? Thanks!